### PR TITLE
!!! TASK: Neos / Flow v7 Compatibility

### PR DESCRIPTION
**Pull request by mficzel**
The redirect http component is replaced by a middleware. 

The failing tests are fixed in #50 which is included here aswell but should be merged to master first.

### CODE REVIEW

**Comment by mficzel** [comment]
This is merely a 1:1 port from http-compoments to middlewares. It should be considered wether instead of looking at the routing parameters it would make sense to look at the returned result and try to create a redirect in case of a 4xx state.

Will adjust the pr if this is preferred.

**Comment by mficzel** [comment]
FYI: Tests fail currently because Flow 7 is not released yet and because of a breaking change in fluid. I expect them to become green once the both are resolved.

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ matrix:
         - php: 7.2
 
 before_install:
-    - export FLOW_TARGET_VERSION=6.0
+    - export FLOW_TARGET_VERSION=6.3
```
Comment (kdambekalns) [inline .travis.yml:11]:
```suggestion
    - export FLOW_TARGET_VERSION=7.0
```
Otherwise it won't work, right?

**Review by Sebobo — APPROVED** [review]
Works on my machine!

**Comment by albe** [comment]
Works on Travis' machine!

**Review by albe — APPROVED** [review]

**Review by bwaidelich — APPROVED** [review]
Looks great to my (by reading)

**Pull request merged**